rclcpp/include/rclcpp/service.hpp

```diff
@@ -116,11 +119,22 @@ class Service : public ServiceBase
     auto service_type_support_handle = get_service_type_support_handle<ServiceT>();
 
     // rcl does the static memory allocation here
-    service_handle_ = new rcl_service_t;
-    *service_handle_ = rcl_get_zero_initialized_service();
+    service_handle_ = std::shared_ptr<rcl_service_t>(
+      new rcl_service_t, [=](rcl_service_t *service)
+      {
+        if (rcl_service_fini(service, node_handle_.get()) != RCL_RET_OK) {
+          RCUTILS_LOG_ERROR_NAMED(
+            "RCLCPP",
```

Comment:
I think the logger names ought to be lowercase.

Reply:
Also, this might be a good candidate for using a `rclcpp` sub logger of the node's logger so that it appears as `my_node.rclcpp` rather than just `rclcpp`. Imagine this error comes out on a process with several nodes in it. Having the node name as context would help you narrow it down.

Reply:
Here and elsewhere in the pr.

Reply:
Ok, I've fixed the casing for RCLCPP => rclcpp. 

Regarding the node sublogger, shouldn't this be a rcutils concerns (as opposed to encoding the node name when writing the log)?

Reply:
Well not if you use the RCLCPP_* logging macros, then it’s rclcpp’s issue. Also how would rcutils know anything about node subloggers. It seems pretty obvious to me that the name needs to be injected by the user.

Reply:
I've done it for this message only. But the code looks ugly to me. I wonder if there is a smarter way to simplify that...

Reply:
If the service had a handle to the C++ node, then it could call `rclcpp::Node::get_logger()`, which would  be slightly cleaner looking, but it does exactly the same thing. I'll try to re-review this asap.